src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -219,30 +226,42 @@ struct nonmaxsuppression
 
     // filter boxes below score_threshold
     template <class T>
-    std::priority_queue<std::pair<double, int64_t>>
+    std::vector<std::pair<double, int64_t>>
     filter_boxes_by_score(T scores_start, std::size_t num_boxes, double score_threshold) const
     {
-        std::priority_queue<std::pair<double, int64_t>> boxes_heap;
+        std::vector<std::pair<double, int64_t>> boxes_heap;
         auto insert_to_boxes_heap =
-            make_function_output_iterator([&](const auto& x) { boxes_heap.push(x); });
+            make_function_output_iterator([&](const auto& x) { boxes_heap.push_back(x); });
```

Comment:
`std::back_inserter` can just be used instead.

Reply:
got it.

Reply:
Done